### PR TITLE
Refactor cleanup systemtest

### PR DESCRIPTION
**Pull request by kyrrek**

### CODE REVIEW

File: neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/ServiceProvider.java [hunk 1]
```diff
@@ -71,6 +71,10 @@ public void setSubscriptions(Set<LocalSubscription> subscriptions) {
 		this.subscriptions = subscriptions;
 	}
 
+	public boolean removeSubscription(LocalSubscription subscription) {
```
Comment (RogerMartinsen) [inline neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/ServiceProvider.java]: Onboard rest controller sets the status. Test what is used in main scope.

File: neighbour-model/src/main/java/no/vegvesen/ixn/federation/transformer/CapabilityTransformer.java [hunk 2]
```diff
@@ -17,10 +19,10 @@ public Capabilities capabilityApiToCapabilities(CapabilityApi capabilityApi) {
 		return capabilities;
 	}
 
-	public CapabilityApi selfToCapabilityApi(Self self) {
+	public CapabilityApi selfToCapabilityApi(String selfName, Set<DataType> localCapabilities) {
```
Comment (RogerMartinsen) [inline neighbour-model/src/main/java/no/vegvesen/ixn/federation/transformer/CapabilityTransformer.java]: Please keep the one parameter Self

File: neighbour-model/src/test/java/no/vegvesen/ixn/federation/repository/ServiceProviderRepositoryIT.java [hunk 3]
```diff
@@ -175,6 +175,28 @@ public void testChangingLocalSubscriptionsWithNewSetAnSeeIfWeGetDeletedOneThatIs
...
+		repository.save(serviceProvider);
+
+		serviceProvider = repository.findByName(name);
+		assertThat(serviceProvider.getSubscriptions()).hasSize(1);
+		LocalSubscription subscription = serviceProvider
+				.getSubscriptions()
+				.stream()
+				.findFirst()
+				.orElseThrow(() -> new AssertionError("could not find first subscription"));
+		serviceProvider.removeSubscription(subscription);
```
Comment (RogerMartinsen) [inline neighbour-model/src/test/java/no/vegvesen/ixn/federation/repository/ServiceProviderRepositoryIT.java]: see comment above, use state TEAR_DOWN

File: neighbour-rest-facade/src/main/java/no/vegvesen/ixn/federation/discoverer/facade/NeighbourRESTFacade.java [hunk 4]
```diff
@@ -36,17 +36,17 @@ public NeighbourRESTFacade(NeighbourRESTClient neighbourRESTClient, CapabilityTr
 	}
 
 
-	public Capabilities postCapabilitiesToCapabilities(Self self, Neighbour neighbour) {
+	public Capabilities postCapabilitiesToCapabilities(Neighbour neighbour, String selfName, Set<DataType> localCapabilities) {
```
Comment (RogerMartinsen) [inline neighbour-rest-facade/src/main/java/no/vegvesen/ixn/federation/discoverer/facade/NeighbourRESTFacade.java]: Keep the Self as one parameter.

File: onboard-service/src/test/java/no/vegvesen/ixn/onboard/SelfServiceIT.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package no.vegvesen.ixn.onboard;
+
```
Comment (RogerMartinsen) [inline onboard-service/src/test/java/no/vegvesen/ixn/onboard/SelfServiceIT.java]: Please remove unused class :)

**Review by RogerMartinsen — CHANGES_REQUESTED** [review]
Nice work! Please change back the Self as one parameter as noted above.

File: neighbour-model/src/main/java/no/vegvesen/ixn/federation/transformer/CapabilityTransformer.java (hunk 2 above)
Comment (kyrrek) [reply neighbour-model/src/main/java/no/vegvesen/ixn/federation/transformer/CapabilityTransformer.java]: Done

File: neighbour-model/src/test/java/no/vegvesen/ixn/federation/repository/ServiceProviderRepositoryIT.java (hunk 3 above)
Comment (kyrrek) [reply neighbour-model/src/test/java/no/vegvesen/ixn/federation/repository/ServiceProviderRepositoryIT.java]: done

File: neighbour-rest-facade/src/main/java/no/vegvesen/ixn/federation/discoverer/facade/NeighbourRESTFacade.java (hunk 4 above)
Comment (kyrrek) [reply neighbour-rest-facade/src/main/java/no/vegvesen/ixn/federation/discoverer/facade/NeighbourRESTFacade.java]: done

**Pull request merged**